packages/testcontainers/src/containers/Hydra/NativeChainContainer.ts

```diff
@@ -0,0 +1,225 @@
+
+  public withStartOptions (startOptions: StartOptions): this {
+    this.startOptions = startOptions
+    return this
+  }
+
+  protected networkConfig: JellyfishNetwork = getNetwork('testnet')
+
+  public withJellyfishNetwork (networkName: NetworkName): this {
+    this.networkConfig = getNetwork(networkName)
+    return this
+  }
+
+  protected cachedRpcUrl?: string
+
+  /**
+   * Convenience Cmd builder with StartOptions
+   */
+  protected getCmd (): string[] {
+    const { user, password } = this.startOptions
+
+    return [
+      'defid',
+      '-printtoconsole',
+      '-rpcallowip=0.0.0.0/0',
+      '-rpcbind=0.0.0.0',
+      '-rpcworkqueue=512',
+      `-rpcuser=${user}`,
+      `-rpcpassword=${password}`
+    ]
+  }
+
+  /* eslint-disable @typescript-eslint/no-non-null-assertion, no-void */
+  public static readonly PREFIX = 'defichain-testcontainers-'
+
+  /**
+   * Create container and start it immediately waiting for NativeChain to be ready
+   */
+  public async start (): Promise<StartedNativeChainContainer> {
+    const network = await new Network().start()
+
+    const rand = Math.floor(Math.random() * 10000000)
+
+    this.withExposedPorts(...Object.values(this.networkConfig.ports))
+      .withName(`${NativeChainContainer.PREFIX}-${this.networkConfig.name}-${rand}`)
+      .withNetworkMode(network.getName())
+      .withCmd(this.getCmd())
+      .withStartupTimeout(120_000)
+
+    const startedContainer = new StartedNativeChainContainer(await super.start(), this.startOptions, this.networkConfig)
+    return startedContainer
+  }
+}
+
+export class StartedNativeChainContainer extends AbstractStartedContainer {
+  // Isaac: is this still necessary?
+  protected cachedRpcUrl?: string
+
+  constructor (
+    startedTestContainer: StartedTestContainer,
+    private readonly startOptions: StartOptions = NativeChainContainer.DefaultStartOptions,
+    private readonly networkConfig: JellyfishNetwork = getNetwork('testnet')
+  ) {
+    super(startedTestContainer)
+  }
+
+  /**
+   * Set contents of ~/.defi/defi.conf
+   * @param {string[]} options to set
+   */
+  async setDeFiConf (options: string[]): Promise<void> {
+    if (options.length > 0) {
+      const fileContents = `${options.join('\n')}\n`
+      await this.exec(['bash', '-c', `echo "${fileContents}" > ~/.defi/defi.conf`])
+    }
+  }
+
+  /**
+  * Get host machine url used for NativeChain rpc calls with auth
+  * TODO(fuxingloh): not a great design when network config changed, the url and ports get refresh
+  */
+  async getCachedRpcUrl (): Promise<string> {
+    if (this.cachedRpcUrl === undefined) {
+      const port = this.getMappedPort(this.networkConfig.ports.rpc)
+      const user = this.startOptions.user
+      const password = this.startOptions.password
+      this.cachedRpcUrl = `http://${user}:${password}@127.0.0.1:${port}/`
+    }
+    return this.cachedRpcUrl
+  }
+
+  async getJsonRpcProvider (): Promise<JsonRpcClient> {
```

Comment:
```suggestion
  async getJsonRpcClient (): Promise<JsonRpcClient> {
```

I would change to `getJsonRpcClient` since "Provider" is more of an EVM standard due to the ability to use different networking protocols for interfacing communication.